packages/components/organisms/f-checkout/src/services/tests/mapper.test.js

```diff
@@ -122,3 +122,72 @@ describe('checkout mapper', () => {
         expect(notesRequest[0].note).toBe(userNote);
     });
 });
+
+describe('mapAnalyticsFieldNames :: ', () => {
+    it.each([
+        ['address.line1', 'addressLine1'],
+        ['line1', 'addressLine1'],
+        ['address.line2', 'addressLine2'],
+        ['line2', 'addressLine2'],
+        ['address.city', 'addressCity'],
+        ['city', 'addressCity'],
+        ['address.postcode', 'addressPostcode'],
+        ['postcode', 'addressPostcode'],
+        ['customer.firstName', 'firstName'],
+        ['customer.lastName', 'lastName'],
+        ['customer.mobileNumber', 'phone'],
+        ['mobileNumber', 'phone'],
+        ['customer.email', 'email']
+    ])('should map the fieldname %s to the analytics value %s', (provided, expected) => {
+        // Act & Assert
+        expect(mapAnalyticsFieldNames(provided)).toEqual(expected);
+    });
+
+    it('should correctly map an array of field names and sort alphabetically', () => {
+        // Arrange
+        const provided = [
+            'city',
+            'address.line1',
+            'customer.email',
+            'customer.firstName',
+            'lastName',
+            'mobilePhone'
+        ];
+
+        const expected = [
+            'addressCity',
+            'addressLine1',
+            'email',
+            'firstName',
+            'lastName',
+            'mobilePhone'
+        ];
+
+        // Act & Assert
+        expect(mapAnalyticsFieldNames(provided)).toEqual(expected);
+    });
+
+    it('should sort corrected field names alphabetically', () => {
```

Comment:
I'm struggling to understand the difference between this test and the one above. Could you please help me? :)

Reply:
It was from a comment from Ash where he wanted to know if the method sorted or not. I thought it would be worth adding, but with the new methods it will probably not be needed

Reply:
Just realised I gave them the same name.